From c1f754a85dfd8da2bb740491e7bc4b8263f15ec8 Mon Sep 17 00:00:00 2001
From: sdinkinx <simonx.dinkin@intel.com>
Date: Thu, 2 Mar 2017 19:01:19 +0200
Subject: [PATCH] extend the editing options in hostapd_cli and wpa_cli

with this flag enabled, there is completion option(using TAB)
to the commands(as in regular shell).
in addition, we can use the history of the command, using the
up-down arrows(as in regular shell).

Signed-off-by: sdinkinx <simonx.dinkin@intel.com>
---
 hostapd/.config        | 4 ++++
 wpa_supplicant/.config | 2 +-
 2 files changed, 5 insertions(+), 1 deletion(-)

diff --git a/hostapd/.config b/hostapd/.config
index f9626eb..121fe4a 100644
--- a/hostapd/.config
+++ b/hostapd/.config
@@ -327,3 +327,7 @@ CONFIG_ACS=y
 # These extentions facilitate efficient use of multiple frequency bands
 # available to the AP and the devices that may associate with it.
 CONFIG_MBO=y
+
+# Include internal line edit mode in hostapd_cli. This can be used to provide
+# limited command line editing and history support.
+CONFIG_WPA_CLI_EDIT=y
diff --git a/wpa_supplicant/.config b/wpa_supplicant/.config
index c1d1d81..8a03b2d 100644
--- a/wpa_supplicant/.config
+++ b/wpa_supplicant/.config
@@ -207,7 +207,7 @@ CONFIG_CTRL_IFACE=y
 
 # Include internal line edit mode in wpa_cli. This can be used as a replacement
 # for GNU Readline to provide limited command line editing and history support.
-#CONFIG_WPA_CLI_EDIT=y
+CONFIG_WPA_CLI_EDIT=y
 
 # Remove debugging code that is printing out debug message to stdout.
 # This can be used to reduce the size of the wpa_supplicant considerably
-- 
2.10.1

